crates/pop-cli/src/commands/call/parachain.rs

```diff
@@ -0,0 +1,813 @@
+		prompt_for_variant_param(api, cli, param)
+	} else if param.is_tuple {
+		prompt_for_tuple_param(api, cli, param)
+	} else {
+		prompt_for_composite_param(api, cli, param)
+	}
+}
+
+// Prompt for the value when is a primitive.
+fn prompt_for_primitive_param(cli: &mut impl Cli, param: &Param) -> Result<String> {
+	Ok(cli
+		.input(format!("Enter the value for the parameter: {}", param.name))
+		.placeholder(&format!("Type required: {}", param.type_name))
+		.interact()?)
+}
+
+// Prompt the user to select the value of the Variant parameter and recursively prompt for nested
+// fields. Output example: Id(5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY) for the Id variant.
+fn prompt_for_variant_param(
+	api: &OnlineClient<SubstrateConfig>,
+	cli: &mut impl Cli,
+	param: &Param,
+) -> Result<String> {
+	let selected_variant = {
+		let mut select = cli.select(format!("Select the value for the parameter: {}", param.name));
+		for option in &param.sub_params {
+			select = select.item(option, &option.name, &option.type_name);
+		}
+		select.interact()?
+	};
+
+	if !selected_variant.sub_params.is_empty() {
+		let mut field_values = Vec::new();
+		for field_arg in &selected_variant.sub_params {
+			let field_value = prompt_for_param(api, cli, field_arg)?;
+			field_values.push(field_value);
+		}
+		Ok(format!("{}({})", selected_variant.name, field_values.join(", ")))
+	} else {
+		Ok(format!("{}()", selected_variant.name.clone()))
+	}
+}
+
+// Recursively prompt the user for all the nested fields in a Composite type.
+fn prompt_for_composite_param(
+	api: &OnlineClient<SubstrateConfig>,
+	cli: &mut impl Cli,
+	param: &Param,
+) -> Result<String> {
+	let mut field_values = Vec::new();
+	for field_arg in &param.sub_params {
+		let field_value = prompt_for_param(api, cli, field_arg)?;
+		// Example: Param { name: "Id", type_name: "AccountId32 ([u8;32])", is_optional: false,
+		// sub_params: [Param { name: "Id", type_name: "[u8;32]", is_optional: false, sub_params:
+		// [], is_variant: false }], is_variant: false }
+		if param.sub_params.len() == 1 && param.name == param.sub_params[0].name {
+			field_values.push(field_value);
+		} else {
+			field_values.push(format!("{}: {}", field_arg.name, field_value));
+		}
+	}
+	if param.sub_params.len() == 1 && param.name == param.sub_params[0].name {
+		Ok(field_values.join(", ").to_string())
+	} else {
+		Ok(format!("{{{}}}", field_values.join(", ")))
+	}
+}
+
+// Recursively prompt the user for the tuple values.
+fn prompt_for_tuple_param(
+	api: &OnlineClient<SubstrateConfig>,
+	cli: &mut impl Cli,
+	param: &Param,
+) -> Result<String> {
+	let mut tuple_values = Vec::new();
+	for tuple_param in param.sub_params.iter() {
+		let tuple_value = prompt_for_param(api, cli, tuple_param)?;
+		tuple_values.push(tuple_value);
+	}
+	Ok(format!("({})", tuple_values.join(", ")))
+}
+
+/// Parser to capitalize the first letter of the pallet name.
+fn parse_pallet_name(name: &str) -> Result<String, String> {
```

Comment:
These two methods are only to help prevent user input errors. They automatically convert the pallet name to start with a capital letter (e.g., nfts becomes Nfts) and ensure the extrinsic name is in lowercase (e.g., Mint becomes mint).